src/scheduler/fifo.c

```diff
@@ -1229,35 +1228,34 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int throughput, schd_e
 	}
 
 	if (!rc) {
-		if (pbs_sd != SIMULATE_SD) {
-			if (rjob->is_shrink_to_fit) {
-				char timebuf[TIMEBUF_SIZE] = {0};
-				rc = 1;
-				/* The job is set to run, update it's walltime only if it is not a foerever job */
-				if (rjob->duration != JOB_INFINITY) {
-					convert_duration_to_str(rjob->duration, timebuf, TIMEBUF_SIZE);
-					rc = update_job_attr(pbs_sd, rjob, ATTR_l, "walltime", timebuf, NULL, UPDATE_NOW);
-				}
-				if (rc > 0) {
-					if (strlen(timebuf) > 0) {
-						char logbuf[MAX_LOG_SIZE];
-						snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
-						schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
-					}
-					if (throughput)
-						rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-					else
-						rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
+		if (rjob->is_shrink_to_fit) {
+			char timebuf[TIMEBUF_SIZE] = {0};
+			rc = 1;
+			/* The job is set to run, update it's walltime only if it is not a foerever job */
+			if (rjob->duration != JOB_INFINITY) {
+				convert_duration_to_str(rjob->duration, timebuf, TIMEBUF_SIZE);
+				rc = update_job_attr(pbs_sd, rjob, ATTR_l, "walltime", timebuf, NULL, UPDATE_NOW);
+			}
+			if (rc > 0) {
+				if (strlen(timebuf) > 0) {
+					char logbuf[MAX_LOG_SIZE];
+					snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
+					schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
 				}
-			} else {
 				if (throughput)
```

Comment:
While you are refactoring this part of the code, you could remove the duplicate if (throughput) chunk, this if-else is executed in both the cases, so you can just remove this chunk of code from here and have it after the end of if (rjob->is_shrink_to_fit)  without the 'else'.